### PR TITLE
chore(*): remove btcd support

### PR DESCRIPTION
**Pull request by Lazar955**
This PR aims to remove support for `btcd `we will be only using `bitcoind` as our backend.

References [issue](https://github.com/babylonlabs-io/vigilante/issues/24).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -66,100 +66,69 @@ This will be later used to retrieve the certificate required for RPC connections
...
-We accomplish that through the btcd `btcctl` utility and the use
+We accomplish that through the `bitcoin-cli` utility and the use
 of the parameters we defined above.
 
 ```shell
-btcctl --simnet --wallet --rpcuser=rpcuser --rpcpass=rpcpass \
-       --rpccert=$TESTNET_PATH/bitcoin/rpc-wallet.cert \
-       generate $NUM_BLOCKS
+bitcoin-cli -chain=regtest \
+      -rpcuser=<rpc_user \
```
Comment (gitferry) [inline README.md:129]:
```suggestion
      -rpcuser=<rpc_user> \
```

File: btcclient/notifier.go [hunk 2]
```diff
@@ -1,22 +1,17 @@
...
 	"github.com/btcsuite/btcd/chaincfg"
-	"github.com/btcsuite/btcd/rpcclient"
 	"github.com/btcsuite/btcwallet/chain"
 	"github.com/lightningnetwork/lnd/blockcache"
 	"github.com/lightningnetwork/lnd/chainntnfs"
 	"github.com/lightningnetwork/lnd/chainntnfs/bitcoindnotify"
-	"github.com/lightningnetwork/lnd/chainntnfs/btcdnotify"
 )
 
 type Btcd struct {
```
Comment (gitferry) [inline btcclient/notifier.go:17]: Can we remove this?

File: config/bitcoin.go [hunk 3]
```diff
@@ -12,27 +12,26 @@ import (
...
-	NetParams         string                    `mapstructure:"net-params"`
-	Username          string                    `mapstructure:"username"`
-	Password          string                    `mapstructure:"password"`
-	ReconnectAttempts int                       `mapstructure:"reconnect-attempts"`
-	BtcBackend        types.SupportedBtcBackend `mapstructure:"btc-backend"`
-	ZmqSeqEndpoint    string                    `mapstructure:"zmq-seq-endpoint"`
-	ZmqBlockEndpoint  string                    `mapstructure:"zmq-block-endpoint"`
-	ZmqTxEndpoint     string                    `mapstructure:"zmq-tx-endpoint"`
+	DisableClientTLS  bool                  `mapstructure:"no-client-tls"`
+	CAFile            string                `mapstructure:"ca-file"`
```
Comment (gitferry) [inline config/bitcoin.go:16]: I believe these are not needed

File: config/bitcoin.go [hunk 4]
```diff
@@ -12,27 +12,26 @@ import (
...
+	Endpoint          string                `mapstructure:"endpoint"`
+	WalletEndpoint    string                `mapstructure:"wallet-endpoint"`
+	WalletPassword    string                `mapstructure:"wallet-password"`
+	WalletName        string                `mapstructure:"wallet-name"`
+	WalletCAFile      string                `mapstructure:"wallet-ca-file"`
```
Comment (gitferry) [inline config/bitcoin.go:21]: This one is not needed

File: config/bitcoin.go
Comment (gitferry) [inline config/bitcoin.go:1]: Seems the functions `ReadCAFile` and `ReadWalletCAFile` can be removed

**Review by gitferry — APPROVED** [review]
Nice work! Some more btcd legacy in comments.

File: config/bitcoin.go [hunk 5]
```diff
@@ -12,27 +12,26 @@ import (
...
-	TargetBlockNum    int64                     `mapstructure:"target-block-num"` // this implies how soon the tx is estimated to be included in a block, e.g., 1 means the tx is estimated to be included in the next block
-	NetParams         string                    `mapstructure:"net-params"`
-	Username          string                    `mapstructure:"username"`
-	Password          string                    `mapstructure:"password"`
-	ReconnectAttempts int                       `mapstructure:"reconnect-attempts"`
-	BtcBackend        types.SupportedBtcBackend `mapstructure:"btc-backend"`
-	ZmqSeqEndpoint    string                    `mapstructure:"zmq-seq-endpoint"`
-	ZmqBlockEndpoint  string                    `mapstructure:"zmq-block-endpoint"`
-	ZmqTxEndpoint     string                    `mapstructure:"zmq-tx-endpoint"`
+	DisableClientTLS  bool                  `mapstructure:"no-client-tls"`
```
Comment (KonradStaniec) [inline config/bitcoin.go:15]: I think this can also be deleted as bitcoind does not support tls by default

File: config/bitcoin.go [hunk 6]
```diff
@@ -12,27 +12,26 @@ import (
...
+	CAFile            string                `mapstructure:"ca-file"`
+	Endpoint          string                `mapstructure:"endpoint"`
```
Comment (KonradStaniec) [inline config/bitcoin.go:17]: Is there now a difference between `endpoint` and `wallet-endpoint` ? or `wallet-endpoint` was supossed to be used to handle situations where there are multiple wallets loaded  i.e the things with `Endpoint + "/wallet/" + WalletName` ?

Comment (Lazar955) [reply config/bitcoin.go:17]: @KonradStaniec yeah, `wallet-endpoint` is not used so I'll be removing it, but we do use `cfg.Endpoint + "/wallet/" + cfg.WalletName`

File: config/bitcoin.go (hunk 4 above)
Comment (Lazar955) [reply config/bitcoin.go]: Nice spot, but we do need `WalletPassword` and `WalletName` see [this](https://github.com/babylonlabs-io/vigilante/blob/137b7db70e5cdc3871210bb3a629f0ead8ab0c3f/btcclient/client_wallet.go#L34) and [this](https://github.com/babylonlabs-io/vigilante/blob/137b7db70e5cdc3871210bb3a629f0ead8ab0c3f/submitter/relayer/relayer.go#L225)

**Review by KonradStaniec — APPROVED** [review]
LGTM!

**Pull request merged**